solver/jobs.go

```diff
@@ -817,8 +817,11 @@ func (s *sharedOp) Exec(ctx context.Context, inputs []Result) (outputs []Result,
 	}
 	flightControlKey := "exec"
 	res, err := s.g.Do(ctx, flightControlKey, func(ctx context.Context) (ret interface{}, retErr error) {
-		if s.execRes != nil || s.execErr != nil {
-			return s.execRes, s.execErr
+		if s.execErr != nil {
+			return nil, s.execErr
+		}
+		if s.execRes != nil {
+			return s.execRes, nil
```

Comment:
It looks like this does not atm cache the `nil` value correctly and `Exec` gets called again in that case. This can be addressed in follow up.

Is the only case where this can return nil when loading an image without any layers?